### PR TITLE
Issue #78 - Checkstyle + javadoc for api module

### PR DESCRIPTION
**Pull request by ljacomet**
* Configure checkstyle to require javadoc on API module
* Fix checkstyle errors by adding javadoc

### CODE REVIEW

File: api/src/main/java/org/ehcache/CacheManager.java [hunk 1]
```diff
@@ -25,10 +25,38 @@
...
+  /**
+   * Creates a {@link Cache} in this {@code CacheManager} according to the specified {@link CacheConfiguration}.
+   *
+   * @param alias the alias under which the cache will be created
+   * @param config the configuration of the cache to create
+   * @param <K> the type of the keys used to access data within this cache
+   * @param <V> the type of the values held within this cache
+   * @return the created and initialized {@link Cache}
+   */
   <K, V> Cache<K, V> createCache(String alias, CacheConfiguration<K, V> config);
```
Comment (timeck) [inline api/src/main/java/org/ehcache/CacheManager.java]: this method throws an exception if the cache already exists I think

File: api/src/main/java/org/ehcache/CacheManager.java [hunk 2]
```diff
@@ -25,10 +25,38 @@
  */
 public interface CacheManager {
 
+  /**
+   * Creates a {@link Cache} in this {@code CacheManager} according to the specified {@link CacheConfiguration}.
```
Comment (timeck) [inline api/src/main/java/org/ehcache/CacheManager.java]: I don't have a lot of javadoc experience really. Is there an argument for/against fully qualifying class names in @link refs?

File: api/src/main/java/org/ehcache/config/CacheConfiguration.java [hunk 3]
```diff
@@ -26,37 +26,43 @@
...
+ * @param <V> the type of the values held within the cache
+ *
  * @author Alex Snaps
  */
 public interface CacheConfiguration<K, V> {
 
   /**
-   * Not sure whether this should be exposed on this interface really
-   * @return unmodifiable collection of service configuration related to this cache
+   * Not sure whether this should be exposed on this interface really.
```
Comment (timeck) [inline api/src/main/java/org/ehcache/config/CacheConfiguration.java]: Not necessarily part of this review but maybe we should take the time to answer this question? @alexsnaps  ?

File: api/src/main/java/org/ehcache/event/CacheEvent.java [hunk 4]
```diff
@@ -19,6 +19,11 @@
 import org.ehcache.Cache;
 
 /**
+ * Class representing an event resulting from a mutative {@link Cache} operation.
```
Comment (timeck) [inline api/src/main/java/org/ehcache/event/CacheEvent.java]: doesn't really matter but you could maybe drop the "Class representing" at the beginning

File: api/src/main/java/org/ehcache/exceptions/SerializerException.java [hunk 5]
```diff
@@ -18,24 +18,44 @@
...
  * @author Ludovic Orban
  */
 public class SerializerException extends RuntimeException {
 
   private static final long serialVersionUID = -4008956327217206643L;
 
+  /**
+   * No args constructor for this exception.
+   */
   public SerializerException() {
```
Comment (timeck) [inline api/src/main/java/org/ehcache/exceptions/SerializerException.java]: god kills a kitten every time we write javadoc to satisfy checkstyle for obvious things like this

File: api/src/main/java/org/ehcache/spi/serialization/SerializationProvider.java [hunk 6]
```diff
@@ -19,10 +19,20 @@
 import org.ehcache.spi.service.ServiceConfiguration;
 
 /**
+ * Interface that allows to provide a {@link Serializer} for a given type.
```
Comment (timeck) [inline api/src/main/java/org/ehcache/spi/serialization/SerializationProvider.java]: "that allows" confuses me a little, maybe not necessary

File: api/src/main/java/org/ehcache/CacheManager.java (hunk 2 above)
Comment (ljacomet) [inline api/src/main/java/org/ehcache/CacheManager.java]: I believe you do not need the _fqcn_ if the class is in the import list. And I tend to use the _fqcn_ when I want to reference a class that is not in the import list.

File: api/src/main/java/org/ehcache/spi/serialization/SerializationProvider.java (hunk 6 above)
Comment (ljacomet) [inline api/src/main/java/org/ehcache/spi/serialization/SerializationProvider.java]: Replaced with: Contract to provide a {@link Serializer} for a given type.

File: api/src/main/java/org/ehcache/exceptions/SerializerException.java (hunk 5 above)
Comment (ljacomet) [inline api/src/main/java/org/ehcache/exceptions/SerializerException.java]: At the same time, not sure I want to handle this via a checkstyle exception

File: api/src/main/java/org/ehcache/event/CacheEvent.java (hunk 4 above)
Comment (ljacomet) [inline api/src/main/java/org/ehcache/event/CacheEvent.java]: Done

File: api/src/main/java/org/ehcache/CacheManager.java (hunk 1 above)
Comment (ljacomet) [inline api/src/main/java/org/ehcache/CacheManager.java]: Indeed and another one if the cache fails to init.

**Comment by ljacomet** [comment]
Added a commit to address @timeck's remarks

**Pull request merged**